### DISCUSSION TITLE
Problem with segmentation

### DISCUSSION THREAD

**Discussion by Cipriano80** [Abe404/root_painter#129] [Q&A]
Good morning everyone, I'm a new user of Rootpainter and I find it very interesting.
I would like to ask why, after training, to analyze 18 images, with 144 training images, made using the source images, the model is not accurate. Let me explain better, at the end of the training the "segmentations" seem excellent, but the actual analysis of the images is quite disappointing. It's possible to correct segmentation after training.
Another problem is that COLAB stops working in the afternoon (I'm writing from Italy, I don't know if it can be useful).

Thanks for your help

**Comment by Abe404** [comment]
Good morning :) 

Thanks for taking the time to try out RootPainter.

> "I would like to ask why, after training, to analyse 18 images, with 144 training images, made using the source images, the model is not accurate. Let me explain better, at the end of the training the "segmentations" seem excellent, but the actual analysis of the images is quite disappointing."

It's difficult to understand exactly what might be causing the problem for you, but my best guess right now is you did not select the best created model to segment your images (or perhaps you are just looking at the intermediate segmentations generated during training).

The workflow for using RootPainter is covered in the [mini guide](https://github.com/Abe404/root_painter/blob/master/docs/mini_guide.md). Once your model is trained to a sufficient accuracy, you should then select the 'segment folder' option from the network menu. I suggest creating a results folder first i.e
project_name/results/model_20_seg and then selecting this as your output folder for your segmentations.

Where model_20 is the name of the model you use to segment your images. Typically you should always segment the images with the model with the highest number from the models folder of your project. The models are created sequentially during training so the one with the highest number should have the best accuracy.

One handy way to check the accuracy of the produced segmentations is the 'extract composites' option from the extras menu, which allows you to see the produced segmentations in combination with the original images.

> " It's possible to correct segmentation after training." 

I think it's worth you check the [mini guide](https://github.com/Abe404/root_painter/blob/master/docs/mini_guide.md).

> "Another problem is that COLAB stops working in the afternoon (I'm writing from Italy, I don't know if it can be useful)."

If using the free Colab then there are some limits on the usage. I think pro colab is still cheap and provides more GPU hours but please double check the terms with google as they change from time to time.

Unfortunately I'm not aware of any free solutions that provide unlimited GPU usage.

I hope that helps. Please let me know how you get on.

Kind regards,
Abraham

**Reply by Cipriano80** [reply]
Thanks so much for the suggestions. I will look at the mini-guide carefully. I know the "extract composites" tool well and this is how I realized the poor accuracy of the model. I think I used the most recent model, because I know it creates them sequentially and I got the one with the most recent date and time. I followed your tutorial on biopores for training.

In any case I'll keep you updated.

**Reply by Abe404** [reply]
Thanks. If you could provide step-by-step instructions on how to reproduce the problem then I could look into it but this would likely require that you share your project (including models), training dataset and original dataset. Otherwise please keep me posted. If all else fails we could have a meeting in the new year.

**Reply by Cipriano80** [reply]
Dear Abraham,
give me a mail address, please.
I will send you an e-mail whit zip.file about all material for simulation.

Thanks in advanced!

**Reply by Abe404** [reply]
My email is available if you click my name on the paper: https://nph.onlinelibrary.wiley.com/doi/full/10.1111/nph.18387
(trying not to post my email more than necessary on the public web).

**Reply by Cipriano80** [reply]
Dear Abram,
following your suggestions, I managed to obtain a very satisfactory root analysis.

Thanks again
